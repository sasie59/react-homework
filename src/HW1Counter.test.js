// import React from 'react';
// import { render } from '@testing-library/react';
// import userEvent from '@testing-library/user-event';
// import HW from './HW1Counter';
// test('should HW1', () => {
//   const { container } = render(<HW />);
//   expect(container).toMatchSnapshot();
// });

// test('should simulate onClick event', () => {
//   const { container } = render(<HW />);
//   const button = container.querySelector('button');
//   userEvent.click(button);
//   expect(button.textContent).toBe('Click Me : 1 ');
//   expect(container).toMatchSnapshot();
// });
import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW1Counter';

test('should onClick event', () => {
  const { container } = render(<HW />);
  userEvent.click(container.querySelector('button'));
  expect(container.querySelector('button').textContent).toBe('Click Me : 1 ');
});

