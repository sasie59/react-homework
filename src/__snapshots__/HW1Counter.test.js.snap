// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW1 1`] = `
<div>
  <div>
    <h1>
      HW1Counter
    </h1>
    <button>
      counter : 
      0
    </button>
  </div>
</div>
`;

exports[`should simulate onClick event 1`] = `
<div>
  <div>
    <h1>
      HW1Counter
    </h1>
    <button>
      counter : 
      1
    </button>
  </div>
</div>
`;
